yunwoo/baekjoon/수학/2292-벌집.js

```diff
@@ -0,0 +1,31 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  const n = +input[0];
+  let result = 1;
+  let roomLength = 1;
+  let i = 1;
+
+  while (n > roomLength) {
+    result++;
+    roomLength += i * 6;
+    i++;
+  }
+
+  console.log(result);
```

Comment:
```suggestion
  const n = +input[0];
  let roomLength = 1;
  let count = 1;

  while (n > roomLength) {
    roomLength += count++ * 6;
  }

  console.log(count);
```
- 풀이 잘봤습니다 🙇‍♂️
- result와 i를 동시에 올리기때문에 둘 중 하나는 필요없는 변수라는 생각이드네요! 둘을 통합해서 count라는 변수로 바꿔줬습니다.
- 그리고 증감 연산자를 뒤에다 붙이면 현재 값을 이용하고 그 다음에 값을 증가시키기 때문에 아래와 같은 형태가 가능합니다.
```js
roomLength += count++ * 6;
```

Reply:
- 앗.. 변수 하나면 충분하네요!!
- 헉 처음알았습니다 주어진 식은 ++이 되기 전 값으로 계산이 되는군요!!! 감사합니다 👍🙇‍♂️